### PR TITLE
linearizer: change globals to merge into left axis/gridDims.x first

### PR DESCRIPTION
**Pull request by flammit**
also fixes Variable max size to be correct and add docs for the off parameter

### CODE REVIEW

File: test/test_linearizer.py [hunk 1]
```diff
@@ -628,22 +628,26 @@ def _assert_grouped_dims(prefix, dims, max_sizes, reverse_dims, expected_sizes):
...
 
+    with self.assertRaises(AssertionError): # dim too large and not factorable
+      get_grouped_dims("gidx", 0, (128,3,4), (16,4,23,), False,)
+
     with self.assertRaises(AssertionError): # too large for sizes
       get_grouped_dims("gidx", 0, (2,3,4,5,6), (16,16,16,), False,)
 
     with self.assertRaises(AssertionError): # variable too large
-      get_grouped_dims("gidx", 0, (Variable("start_pos", 0, 17),3,4), (16,16,16,), False,)
+      get_grouped_dims("gidx", 0, (Variable("start_pos", 0, 16),3,4), (16,16,16,), False,)
```
Comment (chenyuxyz) [inline test/test_linearizer.py:650]: can you add some cases that has Variable and collapsed?

Comment (chenyuxyz) [reply test/test_linearizer.py:650]: oh i see `TODO: support sint properly`, maybe it's not supported yet?

Comment (flammit) [reply test/test_linearizer.py:649]: Oh right it's not supported yet, but I added a test case that passes and one that should when it is.

**Comment by chenyuxyz** [comment]
i think it's fine to merge this, generally i feel this portion can be made simpler

also interested in benchmarking some kernels that hit this to know which way of folding is more performant

**Pull request merged**